presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java

```diff
@@ -84,4 +66,40 @@ public static boolean lessThan(MethodHandle lessThanFunction, Type type, Block l
 
         return leftArray.getPositionCount() < rightArray.getPositionCount();
     }
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    @TypeParameterSpecialization(name = "T", nativeContainerType = long.class)
+    public static boolean lessThanLong(
+            @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"T", "T"}) MethodHandle lessThanFunction,
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") Block leftArray,
+            @SqlType("array(T)") Block rightArray)
+    {
+        int len = Math.min(leftArray.getPositionCount(), rightArray.getPositionCount());
+        int index = 0;
+        while (index < len) {
+            checkElementNotNull(leftArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            checkElementNotNull(rightArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            long leftElement = type.getLong(leftArray, index);
+            long rightElement = type.getLong(rightArray, index);
+            try {
+                if ((boolean) lessThanFunction.invokeExact(leftElement, rightElement)) {
+                    return true;
+                }
+                if ((boolean) lessThanFunction.invokeExact(rightElement, leftElement)) {
```

Comment:
This doesn't look right. `!(a < b)` doesn't necessarily imply `a = b` or `a > b`. Some types may only define a partial order (e.g., a dependency graph or a vector clock). 

So, for instance, if we have these two arrays:

```
A = [a1, a2]
B = [b1, b2, b3]
```

and `a1 = b1`, but `a2` is neither `<`, `>` or `=` to `b2`, the function will incorrectly indicate `A < B`

The implementation should depend on the `=` operator, too, and explicitly check for equality instead of assuming that if `!(x > y)` and `!(x < y)`, then `x = y`.